src/libcore/mem.rs

```diff
@@ -483,6 +483,15 @@ pub fn align_of_val<T: ?Sized>(val: &T) -> usize {
 /// ```
 #[inline]
 #[stable(feature = "needs_drop", since = "1.21.0")]
+#[cfg(not(stage0))]
+pub const fn needs_drop<T>() -> bool {
```

Comment:
The const variant should be initially unstable. Look for `#[rustc_const_unstable()]` elsewhere in the codebase for examples.

Reply:
Added a gate named `const_needs_drop`.